geometry/symmetric_bilinear_form_body.hpp

```diff
@@ -17,13 +19,58 @@ namespace principia {
 namespace geometry {
 namespace internal_symmetric_bilinear_form {
 
+using quantities::Abs;
 using quantities::Angle;
 using quantities::ArcCos;
 using quantities::Cos;
+using quantities::IsFinite;
 using quantities::Sqrt;
 using quantities::Square;
 using quantities::si::Radian;
 
+struct CosSin {
+  double cos;
+  double sin;
+};
+
+// This is J(p, q, θ) in [GV13] section 8.5.1.  This matrix is also called a
+// Givens rotation.
+inline R3x3Matrix<double> JacobiRotation(int const p,
+                                         int const q,
+                                         CosSin const& cos_sin) {
+  R3x3Matrix<double> j = R3x3Matrix<double>::Identity();
+  j(p, p) = cos_sin.cos;
+  j(q, q) = cos_sin.cos;
+  j(p, q) = cos_sin.sin;
+  j(q, p) = -cos_sin.sin;
+  return j;
+};
+
+// See [GV13] section 8.5.2.
+template<typename Scalar>
+CosSin SymmetricShurDecomposition2(R3x3Matrix<Scalar> const& A,
+                                   int const p,
+                                   int const q) {
+  static Scalar const zero{};
+  CosSin cos_sin;
```

Comment:
Rename this to θ for consistency with the surrounding notation in GV13, and use `auto& [c, s] = θ`.